### ISSUE TITLE
Docking: Feature List (RFC)

### ISSUE THREAD

**Issue by christophwille** [icsharpcode/ILSpy#1724]
Multiple Decompilation Tabs

* A new decompilation tab is opened via context menu on tree node, otherwise decompilation is sent to currently active tab. (Nice-to-have would be that right-click ctx menu opening doesn't navigate to that node immediately, because that sort of makes open in new tab duplicate that immediately)
* Last open decompilation tab cannot be closed.
* Each tab retains its own state for language and language version dropdown.
* Tab title is type name (Improvement: option for showing asm name in brackets, or coloring tab by asm similar to VS Productivity Pack did with source tabs by project; could add additional info to tab maybe via tooltip?)

Docking in General

* Arrange decompilation tabs vertical (esp. useful)/horizontal (Use cases: compare stages on same type, compare types across versions)
* Tab context menu (down arrow right-hand side): hide/close depending on context.

**Comment by weltkante** [comment]
I've tried out the docking UI from master branch and one thing I notice is that the minimum height of the search/analyze panels is a bit annoying. I've usually had these panels smaller than what the minimum height enforces now.

**Comment by siegfriedpammer** [comment]
@weltkante can you please test again with the latest master? If the problem is still there, please clarify. Thanks!

**Comment by weltkante** [comment]
seems fixed 👍

**Comment by trivalik** [comment]
I upgraded today from preview 2 to ILSpy 6.0 Preview 3. I was confused that there is a new dock window "Debug ...". I load for test purpose a DLL from GAC, still confused. I closed the debug window (which was still docked) and received endless appearing callstack message boxes. After I restart ILSpy I see no ability to reopen the "Debug..." window.

**Comment by siegfriedpammer** [comment]
Yes, this was a bug, which is fixed in preview 4. The "Debug" window should only  be visible in Debug builds.

**Comment by miloush** [comment]
Once I undock (accidentally) the Search pane, how do I put it back to the place? The code pane is not offering the docking cross UI...

**Comment by siegfriedpammer** [comment]
You can always use the "reset layout" command from the view menu to "repair" the layout. Yes, this is a known limitation of the docking library we are using.
